hivemind/client/averaging/__init__.py

```diff
@@ -291,16 +289,10 @@ async def _step(self, *, future: MPFuture, gather_binary: bytes, weight: float,
                                                                         data_for_gather=data_for_gather)
                     if group_info is None:
                         raise AllreduceException("Averaging step failed: could not find a group.")
-                    group_id = group_info.group_id
-                    allreduce_runner = await self._make_allreduce_runner(group_info, **self.allreduce_kwargs)
-                    self._running_groups[group_id] = allreduce_runner
-                    self._pending_group_assembled.set()
-                    await asyncio.wait_for(allreduce_runner.run(), self._allreduce_timeout)
-                    if self.mode != AveragingMode.AUX:
-                        await loop.run_in_executor(None, self.update_tensors, allreduce_runner)
 
-                    # averaging is finished, exit the loop
-                    future.set_result(allreduce_runner.gathered)
+                    future.set_result(await asyncio.wait_for(
```

Comment:
Please note: the codecov message is a lie :)

Codecov does not collect lines that were used in a separate process. It should feel bad for not doing so.

Reply:
pytest-cov.readthedocs.io/en/latest/subprocess-support.html

Reply:
The trick is, we're deliberately not calling join in some cases.

Though, thois should not apply to DecentralizedAverager

Reply:
created a separate issue for this

https://github.com/learning-at-home/hivemind/issues/293